src/daemon/distribution.c

```diff
@@ -337,14 +344,42 @@ double distribution_squared_deviation_sum(distribution_t *dist) {
   return squared_deviation_sum;
 }
 
+double distribution_stddev(distribution_t *dist) {
+  if (dist == NULL || dist->num_buckets == 0) {
+    errno = EINVAL;
+    return NAN;
+  }
+  if (dist->num_buckets == 1) {
+    return 0.0;
+  }
+  pthread_mutex_lock(&dist->mutex);
+  double stddev = sqrt(((((double)dist->num_buckets) * dist->total_square_sum) -
+                        (dist->total_sum * dist->total_sum)) /
+                       ((double)(dist->num_buckets * (dist->num_buckets - 1))));
+  pthread_mutex_unlock(&dist->mutex);
+  return stddev;
+}
+
+int distribution_reset(distribution_t *dist) {
+  if (dist == NULL) {
+    return errno = EINVAL;
+  }
+  dist->total_sum = 0;
+  dist->total_square_sum = 0;
+  for (size_t i = 0; i < tree_size(dist->num_buckets); i++) {
+    dist->tree[i].bucket_counter = 0;
+  }
+  return 0;
+}
+
 /* TODO(bkjg): add tests for this function */
 int distribution_sub(distribution_t *d1, distribution_t *d2) {
   if (d1 == NULL || d2 == NULL) {
-    return EINVAL;
+    return errno = EINVAL;
```

Comment:
Hm, I was a bit confused with it because when we have functions that has return values then we set errno to smth. In other case when our functions are voids we just return the error code and don't set errno. I thought that it would be better to have one approach to all functions but I'm not sure. If you think that it's odd I'll remove it